File: programme/randmon-prime-number/random-prime-number.py
# Write a programme to generate a random prime number in range n1-n2

import random


def is_prime(n):
    for i in range(2, n):
        if (n % i) == 0:
            return False
    return True


def random_prime_number(number1, number2):
    random_number = random.randint(number1, number2)

    while not is_prime(random_number):
        random_number = random.randint(number1, number2)
    return random_number


range_start = 0
range_end = 100
print("Start range :", range_start)
print("End   range :", range_end)
print("\nRandom prime number :", random_prime_number(range_start, range_end))

File: programme/random-prime-number/random-prime-number.c
// Write a programme to generate a random prime number in range n1-n2

/*
Programm take as given that inputs are always lower<upper
Normally we would need a input control to confirm that the uper argument is always bigger
*/

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int isPrime(int n)
{
  int i;
  if (n == 0 || n == 1)
    return 0;
  for (i = 2; i <= n / 2; i++)
  {
    if (n % i == 0)
    {
      return 0;
    }
  }
  return 1;
}

int main()
{
  int lower, upper, num;

  srand(time(0));
  printf("Enter lower limit : ");
  scanf("%d", &lower);
  printf("Enter upper limit : ");
  scanf("%d", &upper);

  printf("\nThe random number is : ");

  int check = 0;
  while (check != 1)
  {
    num = rand() % (upper - lower + 1) + lower;
    check = isPrime(num);
  }
  printf("%d\n", num);

  return 0;
}
